src/runtime_src/core/tools/xbmgmt2/SubCmdDump.cpp

```diff
@@ -34,22 +37,50 @@ namespace po = boost::program_options;
 // ------ L O C A L   F U N C T I O N S ---------------------------------------
 
 static void
-flash_dump(std::ofstream& /*fOutput*/)
+flash_dump(const std::string& output, const std::shared_ptr<xrt_core::device>& _dev)
 {
   XBU::verbose("Option: flash");
   // Sample output:
   //   Output file: foo.bin
   //   Flash Size: 0x222 (Mbits)  
   //   <Progress Bar>
 
-  // TO-DO
+  Flasher flasher(_dev->get_device_id());
+  if(!flasher.isValid()) {
+    xrt_core::error(boost::str(boost::format("%d is an invalid index") % _dev->get_device_id()));
+    return;
+  }
+  flasher.readBack(output);

```

Comment:
No camelCase please. readBack -> read_back

Reply:
this is just consistent to all methods in flasher.cpp